sklearn/discriminant_analysis.py

```diff
@@ -11,8 +11,8 @@
 
 import warnings
 import numpy as np
+import scipy.linalg
 from scipy import linalg
```

Comment:
For this PR, I only used the full name where we had to select between `xp.linalg` and `scipy.linalg`, which I find more explicit:

https://github.com/thomasjpfan/scikit-learn/blob/5bc6f76f843d7ddf8e458021be0bf8e8f5ce1ad9/sklearn/discriminant_analysis.py#L484-L487

As for the other parts of the file, I would prefer to fully qualify `linalg` because I often need to look at the top of the file to see if it's `numpy.linalg` or `scipy.linalg`. I think this discussion and decision can be done independently of this PR.